src/integrationTest/java/org/opensearch/test/framework/cluster/LocalOpenSearchCluster.java

```diff
@@ -0,0 +1,491 @@
+
+import org.opensearch.action.admin.cluster.health.ClusterHealthResponse;
+import org.opensearch.client.AdminClient;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.health.ClusterHealthStatus;
+import org.opensearch.common.Strings;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.http.BindHttpException;
+import org.opensearch.node.PluginAwareNode;
+import org.opensearch.plugins.Plugin;
+import org.opensearch.test.framework.certificate.TestCertificates;
+import org.opensearch.test.framework.cluster.ClusterManager.NodeSettings;
+import org.opensearch.transport.BindTransportException;
+
+import static java.util.Objects.requireNonNull;
+import static org.junit.Assert.assertEquals;
+import static org.opensearch.test.framework.cluster.NodeType.CLIENT;
+import static org.opensearch.test.framework.cluster.NodeType.DATA;
+import static org.opensearch.test.framework.cluster.NodeType.MASTER;
+
+public class LocalOpenSearchCluster {
+
+    static {
+        System.setProperty("opensearch.enforce.bootstrap.checks", "true");
+    }
+
+    private static final Logger log = LogManager.getLogger(LocalOpenSearchCluster.class);
+
+    private final String clusterName;
+    private final ClusterManager clusterManager;
+    private final NodeSettingsSupplier nodeSettingsSupplier;
+    private final List<Class<? extends Plugin>> additionalPlugins;
+    private final List<Node> nodes = new ArrayList<>();
+    private final TestCertificates testCertificates;
+
+    private File clusterHomeDir;
+    private List<String> seedHosts;
+    private List<String> initialMasterHosts;
+    private int retry = 0;
+    private boolean started;
+    private Random random = new Random();
+
+    public LocalOpenSearchCluster(String clusterName, ClusterManager clusterManager, NodeSettingsSupplier nodeSettingsSupplier,
+                          List<Class<? extends Plugin>> additionalPlugins, TestCertificates testCertificates) {
+        this.clusterName = clusterName;
+        this.clusterManager = clusterManager;
+        this.nodeSettingsSupplier = nodeSettingsSupplier;
+        this.additionalPlugins = additionalPlugins;
+        this.testCertificates = testCertificates;
+        try {
+			this.clusterHomeDir = Files.createTempDirectory("local_cluster_" + clusterName).toFile();
+		} catch (IOException e) {
+			throw new IllegalStateException(e);
+		}        
+    }
+
+    private List<Node> getNodesByType(NodeType nodeType) {
+        return nodes.stream()
+            .filter(currentNode -> currentNode.hasAssignedType(nodeType))
+            .collect(Collectors.toList());
+    }
+
+    private long countNodesByType(NodeType nodeType) {
+        return getNodesByType(nodeType).stream().count();
+    }
+
+    public void start() throws Exception {
+        log.info("Starting {}", clusterName);
+
+        int masterNodeCount = clusterManager.getMasterNodes();
```

Comment:
More instances of the term master, please replace with cluster manager

Reply:
done